libkbfs/block_retrieval_queue.go

```diff
@@ -234,6 +268,30 @@ func (brq *blockRetrievalQueue) Shutdown() {
 	select {
 	case <-brq.doneCh:
 	default:
+		if brq.prefetcher != nil {
+			brq.prefetcher.Shutdown()
+		}
 		close(brq.doneCh)
 	}
 }
+
+// TogglePrefetcher allows upstream components to turn the prefetcher on or off
+func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) error {
+	brq.prefetchMtx.Lock()
+	defer brq.prefetchMtx.Unlock()
+	select {
+	case <-brq.prefetcher.Shutdown():
+	case <-ctx.Done():
+	}
+	if enable {
+		brq.prefetcher = newBlockPrefetcher(brq)
+	}
+	return ctx.Err()
```

Comment:
I think it's clearer to have a `var err` and set it in the select above. It's possible that the select could take the first case, yet the `ctx` is cancelled in the meantime so `Err` here would be non-nil. Harmless in this case, but still seems sketchy.